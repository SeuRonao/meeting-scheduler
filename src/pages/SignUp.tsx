import React, { useState } from "react";
import { Alert, Button, Form, Image, Stack } from "react-bootstrap";
import { useCreateUserWithEmailAndPassword } from "react-firebase-hooks/auth";
import { useTranslation } from "react-i18next";
import { Navigate } from "react-router-dom";
import LoadingSpinner from "../components/common/LoadingSpinner";
import { auth } from "../utils/firebase/firebaseInit";

export default function SignUp() {
  const { t } = useTranslation();
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [validated, setValidated] = useState(false);
  const [createUserWithEmailAndPassword, user, loading, error] =
    useCreateUserWithEmailAndPassword(auth);

  async function handleSubmit(e: React.FormEvent<HTMLFormElement>) {
    e.preventDefault();
    const form = e.currentTarget;
    if (form.checkValidity()) {
      setValidated(true);
      await createUserWithEmailAndPassword(email, password);
    }
  }

  if (user) {
    return <Navigate to="/" />;
  }

  return (
    <>
      <h1 className="text-center my-3">{t("SignUp.title")}</h1>
      <section>
        <Stack direction="horizontal">
          <Image
            className="col-md-6 mx-auto"
            fluid
            src="/images/login.svg"
            alt=""
          />
          <Stack id="enter" className="justify-content-center">
            {error && <Alert variant="danger">{error.message}</Alert>}
            <Form validated={validated} onSubmit={handleSubmit}>
              <Form.Group controlId="formBasicEmail">
                <Form.Label>{t("SignUp.email")}</Form.Label>
                <Form.Control
                  type="email"
                  placeholder={t("SignUp.enter-email")}
                  onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
                    setEmail(e.target.value)
                  }
                  required
                />
                <Form.Text>{t("SignUp.privacy")}</Form.Text>
              </Form.Group>
              <Form.Group className="my-3" controlId="formBasicPassword">
                <Form.Label>{t("SignUp.password")}</Form.Label>
                <Form.Control
                  type="password"
                  placeholder={t("SignUp.password")}
                  onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
                    setPassword(e.target.value)
                  }
                  required
                  minLength={6}
                />
                <Form.Control.Feedback type="invalid">
                  {t("SignUp.password-not-empty")}
                </Form.Control.Feedback>
              </Form.Group>
              <Stack className="align-items-center">
                {loading ? (
                  <LoadingSpinner />
                ) : (
                  <Button variant="primary" type="submit" size="lg">
                    {t("SignUp.title").toUpperCase()}
                  </Button>
                )}
              </Stack>
            </Form>
          </Stack>
        </Stack>
      </section>
    </>
  );
}
