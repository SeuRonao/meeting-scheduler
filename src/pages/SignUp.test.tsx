import { create, act, ReactTestRenderer } from "react-test-renderer";
import SignUp from "./SignUp";

describe("Testing SignUp", () => {
  test("renders correctly", async () => {
    let component: ReactTestRenderer;
    await act(async () => {
      component = create(<SignUp />);
    });
    expect(component!.toJSON()).toMatchSnapshot();
  });
});
