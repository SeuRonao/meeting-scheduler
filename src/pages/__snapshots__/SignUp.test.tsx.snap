// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SignUp renders correctly 1`] = `
Array [
  <h1
    className="text-center my-3"
  >
    title
  </h1>,
  <section>
    <div
      className="hstack"
    >
      <img
        alt=""
        className="col-md-6 mx-auto img-fluid"
        src="/images/login.svg"
      />
      <div
        className="justify-content-center vstack"
        id="enter"
      >
        <form
          className=""
          onSubmit={[Function]}
        >
          <div>
            <label
              className="form-label"
              htmlFor="formBasicEmail"
            >
              email
            </label>
            <input
              className="form-control"
              id="formBasicEmail"
              onChange={[Function]}
              placeholder="enter-email"
              required={true}
              type="email"
            />
            <small
              className="form-text"
            >
              privacy
            </small>
          </div>
          <div
            className="my-3"
          >
            <label
              className="form-label"
              htmlFor="formBasicPassword"
            >
              password
            </label>
            <input
              className="form-control"
              id="formBasicPassword"
              minLength={6}
              onChange={[Function]}
              placeholder="password"
              required={true}
              type="password"
            />
            <div
              className="invalid-feedback"
            >
              password-not-empty
            </div>
          </div>
          <div
            className="align-items-center vstack"
          >
            <button
              className="btn btn-primary btn-lg"
              disabled={false}
              type="submit"
            >
              TITLE
            </button>
          </div>
        </form>
      </div>
    </div>
  </section>,
]
`;
